python/pip_install/pip_repository.bzl

```diff
@@ -474,18 +478,26 @@ def _pip_repository_impl(rctx):
     if rctx.attr.python_interpreter_target:
         config["python_interpreter_target"] = str(rctx.attr.python_interpreter_target)
 
+    # TODO can we get the bzlmod maco_tmpl and the pip
```

Comment:
Instant merge conflict with the cycle machinery although I like the cleanup and would love to see this get down to one template. How do you want to sequence these changes - and apologies that I haven't had time to respond to the change requests on that patch.

Reply:
I am not certain how to get this down to one template. I think we can simplify it some. I’m working on figuring some changes for bzlmod and through those changes I would like to figure out how to get the two code paths closer.

Reply:
> How do you want to sequence these changes - and apologies that I haven't had time to respond to the change requests on that patch.

No worries.  We will have to figure out the merging process.  I don't think I will have more code, and I can merge yours to mine if needs be.  I am working with others on refactoring bzlmod pip, which is why I did this work.  Getting my head around this code base more.

Reply:
@arrdem I think we have even more of a merge conflict now.